src/coreComponents/mesh/generators/NewGhosting.cpp

```diff
@@ -573,47 +661,67 @@ Adjacency buildAdjacency( MeshGraph const & owned,
+
+    // go ahead and add a vector of the correct size to represent the col indices and values for this row of the matrix (not yet populated)
     std::vector< TriGlbIdx > & ind = indices.emplace_back( numEntriesPerRow.back() );
     std::vector< TriScalarInt > & val = values.emplace_back( numEntriesPerRow.back() );
+
+    // populate the data for this row of the matrix
     for( std::size_t i = 0; i < numEdges; ++i )
     {
+      // col index is just obtained from edge global id in EdgeInfo for this edge
       EdgeInfo const & edgeInfo = edges[i];
       ind[i] = convert.fromEdgeGlbIdx( edgeInfo.index );
+      // use a single int for each edge col index to represent if the edge is flipped and position of edge in face
+      // in this case v = 1 + start_node*numEdges + index_in_face (again add 1 so that zero indexing becomes 1 indexing)
       std::size_t const v = 1 + encode< 2 >( numEdges, { edgeInfo.start, i } );
       val[i] = intConv< TriScalarInt >( v );
     }
+    // Add diagonal data
     ind.back() = ownedGlbIdcs.back();
     val.back() = intConv< TriScalarInt >( numEdges );
   }
   for( auto const & [cgi, faces]: owned.c2f )
   {
     // The same comment as for faces and edges applies for cells and faces (see above).
     // The main differences being that
-    // - the `FaceInfo` as a little more information,
-    // - the diagonal stores the type of the cell which conveys the number of face, nodes, ordering...
+    // - the `FaceInfo` as a little more information, 
+    //     (recall FaceInfo is a struct which holds the global face index, the starting node index, and boolean telling which direction the nodes should be traversed)
+    // - the diagonal stores the type of the cell which conveys the number of face, nodes, ordering... (note this is still a TODO, right now its just numFaces)
     std::size_t const numFaces = std::size( faces );
 
+    // This rank owns this row of the matrix
     ownedGlbIdcs.emplace_back( convert.fromCellGlbIdx( cgi ) );
 
     numEntriesPerRow.push_back( numFaces + 1 );  // `+1` comes from the diagonal
+
+    // go ahead and add a vector of the correct size to represent the col indices and values for this row of the matrix (not yet populated)
     std::vector< TriGlbIdx > & ind = indices.emplace_back( numEntriesPerRow.back() );
     std::vector< TriScalarInt > & val = values.emplace_back( numEntriesPerRow.back() );
+
+    // populate the data for this row of the matrix
     for( std::size_t i = 0; i < numFaces; ++i )
     {
+      // col index is just gotten from the global id in FaceInfo for this face
       FaceInfo const & faceInfo = faces[i];
       ind[i] = convert.fromFaceGlbIdx( faceInfo.index );
+
+      // Encode data with a single int like above, but now int tells us start_node, direction (flipped), and position of face in cell
+      // in this case v = 1 + (isFlipped*numFaces + start_index)*numFaces + index_in_cell (again add 1 so that zero indexing becomes 1 indexing)
```

Comment:
We add one to be sure it's never `0` and is always "visible" and never "optimized out".